src/NServiceBus.AcceptanceTesting/ScenarioWithContext.cs

```diff
@@ -30,25 +30,28 @@ public Task<TContext> Run(TimeSpan? testExecutionTimeout = null)
             var scenarioContext = new TContext();
             contextInitializer(scenarioContext);
 
-            var runDescriptor = new RunDescriptor
-            {
-                ScenarioContext = scenarioContext
-            };
-            runDescriptor.Settings.Merge(settings);
+            var runDescriptor = new RunDescriptor(settings, scenarioContext);
 
             LogManager.UseFactory(new ContextAppenderFactory());
 
             var sw = new Stopwatch();
 
             sw.Start();
-            await ScenarioRunner.Run(runDescriptor, behaviors, done).ConfigureAwait(false);
+            var result = await ScenarioRunner.Run(runDescriptor, behaviors, done).ConfigureAwait(false);
             sw.Stop();
 
+            await runDescriptor.RaiseOnTestCompleted(result);
+
+            DisplayRunResult(result);
             Console.WriteLine("Total time for testrun: {0}", sw.Elapsed);
 
-            return (TContext)runDescriptor.ScenarioContext;
-        }
+            if (result.Result.Failed)
+            {
+                throw result.Result.Exception;
+            }
 
+            return (TContext) runDescriptor.ScenarioContext;
```

Comment:
I see formatting introduced a space here, which I wish it didn't do. I guess we still need to adjust the R# rules for this?